TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/CapturedTrafficToHttpTransactionAccumulator.java

```diff
@@ -38,19 +37,30 @@
 @Slf4j
 public class CapturedTrafficToHttpTransactionAccumulator {
 
-    private final TrafficStreamMap liveStreams;
+    public static final Duration EXPIRATION_GRANULARITY = Duration.ofSeconds(1);
+    private final ExpiringTrafficStreamMap liveStreams;
     private final Consumer<HttpMessageAndTimestamp> requestHandler;
     private final Consumer<RequestResponsePacketPair> fullDataHandler;
 
-    public CapturedTrafficToHttpTransactionAccumulator(Consumer<HttpMessageAndTimestamp> requestReceivedHandler,
+    public CapturedTrafficToHttpTransactionAccumulator(Duration minTimeout,
+                                                       Consumer<HttpMessageAndTimestamp> requestReceivedHandler,
                                                        Consumer<RequestResponsePacketPair> fullDataHandler) {
-        liveStreams = new TrafficStreamMap();
+        liveStreams = new ExpiringTrafficStreamMap(minTimeout, EXPIRATION_GRANULARITY,
+                new ExpiringTrafficStreamMap.BehavioralPolicy() {
+                    @Override
+                    public void onExpireAccumulation(String partitionId,
+                                                     String connectionId,
+                                                     Accumulation accumulation) {
+                        log.error("firing accumulation for accum=[" + connectionId + "]=" + accumulation);
+                        fireAccumulationsCallbacks(accumulation);
```

Comment:
I need to dive into the ExpiringTrafficStreamMap but is this essentially saying that on expiry we will start processing whatever is currently in the accumulation, may be a full message, partial message, or empty. Do we have logic in our Netty pipeline to handle an empty or partial message? I guess I'm assuming that at some point we log that we are unable to process the current message and clear it from our tracking

Reply:
If a message can't be parsed, we'll give up on the transformation pipeline & send it exactly as we got it.